rules/no-reduce.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const methodSelector = require('./utils/method-selector');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const message = 'Array.reduce not allowed';
+
+const PROTOTYPE_SELECTOR = [
+	methodSelector({names: ['call', 'apply']}),
+	'[callee.object.type="MemberExpression"]',
+	`:matches(${
+		[
+			// `[].reduce`
+			[
+				'[callee.object.object.type="ArrayExpression"]',
+				'[callee.object.object.elements.length=0]'
+			],
+			// `Array.prototype.reduce`
+			[
+				'[callee.object.object.object.type="Identifier"]',
+				'[callee.object.object.object.name="Array"]',
+				'[callee.object.object.property.name="prototype"]'
+			]
+		].map(selector => selector.join('')).join(', ')
+	})`,
+	'[callee.object.computed=false]',
+	'[callee.object.property.type="Identifier"]',
+	'[callee.object.property.name="reduce"]'
+].join('');
+
+const METHOD_SELECTOR = [
+	methodSelector({name: 'reduce', min: 1, max: 2}),
+	'[callee.object.property.name!="call"]'
```

Comment:
why?

Reply:
it fails some tests if this is removed
```
a.b.call.reduce(() => {})
```

Reply:
But `a.b.call` should be a array, otherwise there is no `reduce` method.